src/init.cpp

```diff
@@ -1507,6 +1508,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
             fLoaded = true;
         } while(false);
 
+        if (fRequestShutdown) break;
```

Comment:
Without this change, the node shuts down quickly, but following appears (when metrics is enabled):
```
( ... zcash and heart logos ...)
Since starting this node 19 seconds ago:
- You have validated no transactions.

Messages:
- : Error loading block database.
Please restart with -reindex to recover.

Init message: Loading block index...

[Press Ctrl+C to exit] [Set 'showmetrics=0' to hide]
$ 
```
Contrary to the message, it's not necessary to `-reindex`.

Reply:
There's a race condition here: if there was a failure for some other reason than a requested shutdown but there is *also* a requested shutdown by the time execution reaches this point, then the error won't be displayed. I think this fixes it:

```suggestion
        if (fRequestShutdown && strLoadError == "") break;
```